src/parser/content-parser.php

```diff
@@ -233,10 +233,16 @@ protected function source_block( $block, $registered_blocks, $filter_options ) {
 		$sourced_block = [
 			'name'       => $block_name,
 			'attributes' => $block_attributes,
+			'id'         => wp_unique_id(),
 		];
 
+		if ( isset( $filter_options['parentId'] ) ) {
+			$sourced_block['parentId'] = $filter_options['parentId'];
+		}
+
 		if ( isset( $block['innerBlocks'] ) ) {
-			$inner_blocks = array_map( function ( $block ) use ( $registered_blocks, $filter_options ) {
+			$filter_options['parentId'] = $sourced_block['id'];
```

Comment:
The ID of a block is set as the parent ID on its children by passing it down in the filter itself.